addons/mail/models/mail_message_translation.py

```diff
@@ -0,0 +1,38 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+from dateutil.relativedelta import relativedelta
+
+
+class MessageTranslation(models.Model):
+    _name = "mail.message.translation"
+    _description = "Message Translation"
+
+    message_id = fields.Many2one(
+        "mail.message",
+        "Message",
+        required=True,
+        ondelete="cascade",
+    )
+    target_lang = fields.Char(
+        "Target Language",
+        help="Shortened language code used as the target for the translation request.",
+    )
+    source_lang = fields.Char("Source Language", help="Result of the language detection based on its content.")
+    body = fields.Html(
+        "Translation Body",
+        default="",
+        sanitize_style=True,
+        help="String received from the translation request.",
+    )
+
+    def init(self):
+        self.env.cr.execute(
+            f"CREATE UNIQUE INDEX IF NOT EXISTS mail_message_translation_unique ON {self._table} (message_id, source_lang) WHERE message_id IS NOT NULL"
```

Comment:
`WHERE message_id IS NOT NULL` looks overkill as message is supposed to be required

Reply:
The unique should rather be on `message_id` and `target_lang`, as the source will be common to all of them.

Reply:
Indeed, that was dumb